singer_sdk/sinks/sql.py

```diff
@@ -336,6 +399,14 @@ def bulk_insert_records(
         with self.connector._connect() as conn, conn.begin():
             result = conn.execute(insert_sql, new_records)
 
+        # Finish Line for max_size perf counter
+        if self.sink_timer is not None:
+            if self.sink_timer.start_time is not None:
+                self.sink_timer.stop()
+                self.batch_size_rows = self.sink_timer.counter_based_max_size()
+
+            # Starting Line for max_size perf counter
+            self.sink_timer.start()
```

Comment:
Wdyt about moving this up the stack? Perhaps to https://github.com/meltano/sdk/blob/f6bbf0c5ddba689ee1ab6df1f110529f35adf12f/singer_sdk/target_base.py#L498

That way all targets, not just SQL, benefit from this.

Reply:
I like it.  The code has been moved up.  Please let me know if this matches up with what you had in mind.